src/Mvc/Mvc.NewtonsoftJson/src/MvcNewtonsoftJsonOptions.cs

```diff
@@ -49,6 +50,15 @@ public class MvcNewtonsoftJsonOptions : IEnumerable<ICompatibilitySwitch>
         /// <value>Defaults to 30Kb.</value>
         public int InputFormatterMemoryBufferThreshold { get; set; } = 1024 * 30;
 
+        /// <summary>
+        /// Gets or sets a flag to determine whether the value of <see cref="CultureInfo.CurrentCulture"/>
+        /// for the current HTTP request is used for JSON deserialization by <see cref="NewtonsoftJsonInputFormatter"/>.
+        /// </summary>
+        /// <value>
+        /// The default value is <see langword="false"/>.
+        /// </value>
+        public bool ReadJsonWithRequestCulture { get; set; }
```

Comment:
Perhaps this might be a better name if this worked for both input and output formatting?

```suggestion
        public bool UseCurrentCulture { get; set; }
```

@martincostello can this also use the culture for output formatting? Seems weird it only works for reading, but not for writing.

Reply:
Yep that makes sense. Initially I just went down the route outlined in the initial issue, but I agree it would be better to have it be bi-directional.

I'll go with that suggested name for now and look at having the output formatter use it too.

Reply:
So from playing around with this locally, the serializer always writes in ISO format for `DateTime` values (or whatever is configured by the `JsonSerializerSettings`, e.g. the `DateFormatString` property).

Setting the writer/serializer to use the current culture has no effect on this and I'm not sure there's a sensible default to overwrite it for or that it's a good idea to special-case for `DateTime`/`DateTimeOffset`.

I'd also have to introduce a new constructor (or make a breaking change) to allow the `MvcNewtonsoftJsonOptions` to be passed into the output formatter to observe the value of the option to know if to set the culture or not.

Thoughts on best route forward?

Reply:
cc @JamesNK in case he has thoughts

Reply:
I think the main place culture is used with Newtonsoft.Json is when deserializing strings into .NET types.

For example:
```json
{"Amount": "1.0"}
```
```cshar
public class Product
{
    public decimal Amount { get; set; }
}
```

By default the culture is `InvariantCulture`. Changing the culture can be used in apps that want to allow strings like `"1,0"` (seems to be a European thing) to successfully convert to a decimal/double/float.

Reply:
Semi-related to this is how we want to handle culture for simple minimal action parameters (#32377). First we need to default to `InvariantCulture`, but I think we might want a way to configure this.